meilisearch/client.py

```diff
@@ -464,3 +468,67 @@ def wait_for_task(
             An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
         """
         return wait_for_task(self.config, uid, timeout_in_ms, interval_in_ms)
+
+    def generate_tenant_token(
+        self,
+        search_rules: Dict[str, Any],
+        expires_at: Optional[int] = None,
+        api_key: Optional[str] = None
+    ) -> str:
+        """Generate a JWT token for the use of multitenancy.
+
+        Parameters
+        ----------
+        search_rules:
+            A Dictionary or an object which contains the rules to be enforced at search time for all or specific
+            accessible indexes for the signing API Key.
+            In the specific case of you want to have any restrictions you can also use a array ["*"].
+        expires_at (optional):
+            Date and time when the key will expire.
+            Note that if an expires_at value is included it should a `timestamp`.
+        api_key (optional):
+            The API key parent of the token. If you let it empty the client API Key will be used.
+
+        Returns
+        -------
+        jwt_token:
+           A string containing the jwt tenant token.
+           Note: If your token does not work remember that the searchrules is madatory and should be well formatted.
+           `exp` must be a timestamp in the future.
+        """
+        # Standard JWT header for encryption with SHA256/HS256 algorithm
+        header = {
+            "typ": "JWT",
+            "alg": "HS256"
+        }
+
+        api_key = str(self.config.api_key) if api_key is None else str(api_key)
+
+        # Add the required fields to the payload
+        payload = {
+            'apiKeyPrefix': api_key[0:8],
+            'searchRules': search_rules,
+            'exp': expires_at if expires_at is not None else None
```

Comment:
`'exp': expires_at` is all that is needed, This will give the same result so you don't need to do the check for `None`.